grafana/resource_data_source.go

```diff
@@ -365,13 +371,35 @@ func makeJSONData(d *schema.ResourceData) gapi.JSONData {
 }
 
 func makeSecureJSONData(d *schema.ResourceData) gapi.SecureJSONData {
-	return gapi.SecureJSONData{
-		AccessKey:         d.Get("secure_json_data.0.access_key").(string),
-		BasicAuthPassword: d.Get("secure_json_data.0.basic_auth_password").(string),
-		Password:          d.Get("secure_json_data.0.password").(string),
-		SecretKey:         d.Get("secure_json_data.0.secret_key").(string),
-		TlsCACert:         d.Get("secure_json_data.0.tls_ca_cert").(string),
-		TlsClientCert:     d.Get("secure_json_data.0.tls_client_cert").(string),
-		TlsClientKey:      d.Get("secure_json_data.0.tls_client_key").(string),
+	secureJSONData := gapi.SecureJSONData{}
+	if v, ok := d.GetOk("secure_json_data"); ok {
```

Comment:
I'm not sure this is necessary, why these changes?

The go-grafana-api won't include any of these fields if they're set to default values.

Reply:
wasn't sure if the struct's fields should be set to empty strings if the API will omit them in the long run, but i'll revert to keep the same pattern as before 👍